// @linterman/shared-types/src/index.js
import { z } from 'zod';
// ============================================================================
// Configuration
// ============================================================================
export const LintConfigSchema = z.object({
    local_only: z.boolean().default(true),
    rules: z.array(z.string()).optional(),
    saas_url: z.string().url().optional(),
    api_key: z.string().optional(),
    fix: z.boolean().default(false),
});
// ============================================================================
// Severity
// ============================================================================
export const SeveritySchema = z.enum(['error', 'warning', 'info']);
// ============================================================================
// Issue
// ============================================================================
export const LintIssueSchema = z.object({
    rule_id: z.string(),
    severity: SeveritySchema,
    message: z.string(),
    path: z.string(),
    line: z.number().nullable().optional(),
    fix: z.any().nullable().optional(), // JSON pour la correction automatique
});
// ============================================================================
// Stats
// ============================================================================
export const LintStatsSchema = z.object({
    total_requests: z.number(),
    total_tests: z.number(),
    total_folders: z.number(),
    errors: z.number(),
    warnings: z.number(),
    infos: z.number(),
});
// ============================================================================
// Result
// ============================================================================
export const LintResultSchema = z.object({
    score: z.number().min(0).max(100),
    issues: z.array(LintIssueSchema),
    stats: LintStatsSchema,
    timestamp: z.number().optional(),
});
// ============================================================================
// Rule Metadata
// ============================================================================
export const RuleMetadataSchema = z.object({
    id: z.string(),
    name: z.string(),
    category: z.enum(['testing', 'documentation', 'performance', 'structure', 'security', 'api-design']),
    severity: SeveritySchema,
    description: z.string(),
    enabled: z.boolean().default(true),
    custom: z.boolean().default(false),
});
// ============================================================================
// Exports
// ============================================================================
export { 
// Schemas pour validation runtime
LintConfigSchema as ConfigSchema, LintIssueSchema as IssueSchema, LintResultSchema as ResultSchema, LintStatsSchema as StatsSchema, RuleMetadataSchema as RuleSchema, };
